ci-operator/step-registry/workers-rhel/pre-hook/gcp/workers-rhel-pre-hook-gcp-commands.sh

```diff
@@ -16,16 +16,48 @@ if ! whoami &> /dev/null; then
     fi
 fi
 
-cat > scaleup-pre-hook-gcp.yaml <<- 'EOF'
+if ! test -f "${SHARED_DIR}/proxy_private_url"; then
+  echo "Failed to find bastion proxy info, abort." && exit 1
+fi
+proxy_private_url="$(< ${SHARED_DIR}/proxy_private_url)"
+proxy="http://$(echo ${proxy_private_url} | cut -d@ -f2)"
+proxy_username="$(echo ${proxy_private_url} | cut -d@ -f1 | cut -d\/ -f3 | cut -d: -f1)"
+proxy_password="$(echo ${proxy_private_url} | cut -d@ -f1 | cut -d\/ -f3 | cut -d: -f2)"
+
+cat > scaleup-pre-hook-gcp.yaml << EOF
 - name: Configure RHEL machine on GCP
   hosts: new_workers
   any_errors_fatal: true
-  gather_facts: false
+  gather_facts: true
 
   tasks:
+  - name: insert proxy settings into dnf.conf
```

Comment:
seem like this step is duplicated with `workers-rhel-pre-hook-sshtunnel`, is this step is used for `yum makecache` to get internet connectivity? This reminds me about https://redhat-internal.slack.com/archives/CF8SMALS1/p1660638167899109?thread_ts=1660025494.900689&cid=CF8SMALS1

Reply:
@jianlinliu Yes, the proxy setting should take effects on any subsequent `dnf` operations including `dnf makecache`. In fact, the RHEL instances under question should have Internet access already. I tried the proxy way because I'd ever met below scenario once, i.e. when the normal RHEL instance complains on failing to get repo metadata, I launched another RHEL instance with public IP and the `dnf makecache` on it did succeed every time that time. As the error appears time to time, I'm not sure yet when it'll happen. My suggestion for now is, enable the temporary solution and keep watching for some days, and see if it'll help a bit or not.  Thanks!

Reply:
> when the normal RHEL instance complains on failing to get repo metadata
Is this - https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/35188/rehearse-35188-periodic-ci-openshift-verification-tests-master-installer-rehearse-4.10-installer-rehearse-gcp/1614153607173640192/artifacts/installer-rehearse-gcp/workers-rhel-pre-hook-gcp/build-log.txt the failing case?

Reply:
Yes, I mean the error "Error: Failed to download metadata for repo 'rhui-rhel-8-for-x86_64-appstream-rhui-debug-rpms': Cannot download repomd.xml: Cannot download repodata/repomd.xml: All mirrors were tried".

Reply:
In the log, `insert proxy settings into dnf.conf` was already added, but still hit "Error: Failed to download metadata for repo 'rhui-rhel-8-for-x86_64-appstream-rhui-debug-rpms'" issue, I guess need to disable `debug-rpms` repo together, and drop "insert proxy settings into dnf.conf" step.